### PR TITLE
Revert "Change the regex used for domain validation"

### LINKED ISSUE #3046: Core domain name validation
@DL6ER You are right, I expect to be able to add the domain via CLI too.

On a quick look I couldn't find the validation check for the web gui, because I wanted to compare them due to different behavior.

Check out  http://regexr.com/3au3g for an alternative validation check for [list.sh](https://github.com/pi-hole/pi-hole/blob/development/advanced/Scripts/list.sh#L95)

_Originally posted by @Th3M3 in https://github.com/pi-hole/AdminLTE/pull/1078#issuecomment-565761008_

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:** 

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes, and have included unit tests where possible.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

Fix #3046

**How does this PR accomplish the above?:**

Revert commit 76460f0

**What documentation changes (if any) are needed to support this PR?:**

None

### CODE REVIEW

**Comment by WaLLy3K** [comment]
The main purpose of the domain validation logic (as far as memory serves) is to prevent users from adding hostnames/domains that when parsed by FTL (`dnsmasq`), cause unexpected results such as crashing due to the entry being incorrectly formatted.

Basically, the regex causes valid domains from not being added, that's a problem. If it allows `blah123` to be added, then that's fine. 😄

**Comment by DL6ER** [comment]
Background knowledge summary:
Compared to the first days of Pi-hole, we have made some great progress. Pi-hole v5.0 will not load any of the gravity or list domains into its cache. This will also allow for using hundreds of millions of domains without much trouble even on a Pi Zero*.
We store all the domains in a [B-tree](https://en.wikipedia.org/wiki/B-tree) (B as in "balanced") which is also why the run times of `pihole -g` got somewhat longer.

What I actually wanted to write:
We can store whatever we want in the lists. You could even store binary data like images in there if you like*. What will happen is that FTL will look up the queried domain in this B-tree. When a match is found, FTL acts accordingly (block or not). If not, then the domain is on no list and permitted.

**TL;DR**: You can throw whatever you like at the database, you can never corrupt FTL with this.

---

*) Even though it is possible, I don't consider this a good idea!
